articles/implementing-aws-s3-functionalities-on-a-nodejs-graphql-api/index.md

```diff
@@ -0,0 +1,624 @@
+        }
+    };
+
+    objectKeys.forEach((objectKey) => params.Delete.Objects.push({
+        Key:objectKey
+    }));
+
+    let removeObjects = promisify(this.s3.deleteObjects.bind(this.s3));
+
+    await removeObjects(params).catch(console.log);
+
+    return {
+        success:true,
+        message:"Successfully deleted objects"
+    };
+
+};
+```
+
+From above:
+
+- Create a params object which encompasses the name of the bucket to be deleted under `Bucket` key, and an array of `Objects` to be deleted under the `Delete` key.
+
+- Loop through the `objectKeys` passed populating the `Objects` array of the `Delete` key in the `params` object.
+
+- Promisify the `deleteObjects()` function from s3 so that we can use the *async/await*  syntax.
+
+- Delete the objects.
+
+- Send back a response as per the schema.
+
+To test this:
+
+- Ensure that the development server is up and running.
+
+- Head over to Altair GraphQl Client, open a separate tab and paste in the following mutation in the workspace:
+
+```javascript
+mutation DeleteObjects {
+  deleteObjects(
+    bucketName:"simple-image-upload-bucket",
+    objectKeys:["enter_object_key","enter_object_key"]
+  ){
+    success
+    message
+  }
+}
+```
+
+- From the tab of fetching objects, copy and paste at least two object keys to the `objectKeys` array. If you have more objects, feel free to include more.
+
+- In case you are using a different bucket name, make sure you change it.
+
+- Hit the play button and examine the results.
+
+### Deleting an s3 bucket
+
+In case an s3 bucket is of no use anymore, you can always delete it. But to delete an s3 bucket, you have to make sure that there are no objects in it. Else it will fail.
+
+In the same file, under `deleteBucket()` function, we implement the functionality of deleting a bucket:
+
+```javascript
+//delete bucket.
+async deleteBucket(bucketName){
+
+    const params = {
+        Bucket:bucketName
+    };
+
+    let removeBucket = promisify(this.s3.deleteBucket.bind(this.s3));
+
+    await removeBucket(params).catch(console.log);
+
+    return {
+        success:true,
+        message:"Successfully deleted bucket"
+    }
+
+};
+```
+From above:
+
+- Creating a params object which holds the name of the bucket to be deleted under `Bucket` key.
+
+- Promisify the `deleteBucket()` method so that we can use the *async/await* syntax.
+
+- Remove the bucket and send back a response.
+
+To test this:
+
+- Ensure that the development server is running.
+
+- Head over to Altair GraphQl Client, open a separate tab, and paste in the following mutation in the workspace:
+
+```javascript
+mutation DeleteBucket {
+  deleteBucket(bucketName:"enter-a-bucket-name"){
+    success
+    message
+  }
+}
+```
+
+- If you have one bucket, and it has files, go through [creating an s3 bucket](#creating-an-s3-bucket) and create a dummy bucket.
+
+- In the mutation, paste in its name in the `bucketName` value.
+
+- Hit the play button and examine the results.
+
```

Comment:
Please add a new headng called, "Summary" and give a quick summary of what all we did. It would help the reader recall the process.